kythe/cxx/verifier/assertions_to_souffle.cc

```diff
@@ -294,19 +298,31 @@ bool SouffleProgram::Lower(const SymbolTable& symbol_table,
       }
       auto id = FindEVar(i.evar);
       if (inspected_vars.insert(id).second) {
-        absl::StrAppend(&result_spec, result_spec.empty() ? "" : ", ", "v", id);
-        absl::StrAppend(&result_tyspec, result_tyspec.empty() ? "" : ", ", "v",
-                        id, ":",
+        if (result_spec.empty()) {
+          result_spec = "rrec : result_ty";
+          result_argspec = "rrec";
+        }
+        absl::StrAppend(&result_clause,
+                        result_clause.empty() ? ", rrec=[" : ", ", "v", id);
+        absl::StrAppend(&result_tyspec,
+                        result_tyspec.empty() ? ".type result_ty = [" : ", ",
+                        "rv", id, ":",
                         type->second == EVarType::kVName ? "vname" : "number");
+
         inspections_.push_back(i.evar);
       }
     }
+    if (!result_tyspec.empty()) {
+      absl::StrAppend(&result_tyspec, "]");
```

Comment:
So it looks like the corresponding `[` do not _both_ depend on `result_tyspec`; one does, but the other on `result_clause`. Maybe the logic lines up; but it would be nice if that were clearer.